src/overlays/actors/ovl_En_Ru1/z_en_ru1.c

```diff
@@ -19,417 +135,2276 @@ const ActorInit En_Ru1_InitVars = {
+            this->unk_350 = 1;
+            func_80AEE02C(this);
+            this->unk_35C = 0;
+            this->unk_358 = (this->actor.unk_84 - 10.0f) * 0.5f;
+            this->unk_354 = this->actor.posRot.pos.y + thisx->unk_358; // thisx only used here
+        } else {
+            this->actor.gravity = 0.0f;
+            this->actor.minVelocityY *= 0.2f;
+            this->actor.velocity.y *= 0.2f;
+            if (this->actor.minVelocityY >= -0.1f) {
+                this->actor.minVelocityY = 0.0f;
+                this->actor.velocity.y = 0.0f;
+            }
+            this->actor.speedXZ *= 0.5f;
+            if (this->actor.speedXZ <= 0.1f) {
+                this->actor.speedXZ = 0.0f;
+            }
+            this->actor.velocity.x = Math_Sins(this->actor.posRot.rot.y) * this->actor.speedXZ;
+            this->actor.velocity.z = Math_Coss(this->actor.posRot.rot.y) * this->actor.speedXZ;
+            func_8002D7EC(this);
+        }
+    } else {
+        if (this->unk_350 == 1) {
+            if (this->unk_358 <= 1.0f) {
+                func_80AEE02C(this);
+                this->unk_350 = 2;
+                this->unk_360 = 0.0f;
+            } else {
+                sp28 = this->unk_358;
+                sp24 = this->unk_354;
+                temp_f10 = Math_Coss(this->unk_35C) * -sp28;
+                this->actor.posRot.pos.y = temp_f10 + sp24;
+                this->unk_35C += 0x3E8;
+                this->unk_358 *= 0.95f;
+            }
+        } else {
+            this->unk_360 += 1.0f;
+            if (this->unk_360 > 0.0f) {
+                this->unk_350 = 3;
+            }
+        }
+    }
+}
+
+s32 func_80AEE264(EnRu1* this, GlobalContext* globalCtx) {
+    Actor* thisx = &this->actor;
+
+    if (!func_8002F194(thisx, globalCtx)) {
+        thisx->flags |= 9;
+        if ((gSaveContext.inf_table[20] & 8)) {
+            thisx->textId = 0x404E;
+            func_8002F2F4(thisx, globalCtx);
+        } else {
+            if ((gSaveContext.inf_table[20] & 4)) {
+                thisx->textId = 0x404D;
+                func_8002F2F4(thisx, globalCtx);
+            } else {
+                thisx->textId = 0x404C;
+                func_8002F2F4(thisx, globalCtx);
+            }
+        }
+        return 0;
+    }
+    return 1;
+}
+
+void func_80AEE2F8(EnRu1* this, GlobalContext* globalCtx) {
+    DynaPolyActor* dyna;
+    u32 floorPolySource;
+    if ((this->actor.bgCheckFlags & 1) && (this->actor.floorPolySource != 0x32)) {
+        floorPolySource = this->actor.floorPolySource;
+        dyna = DynaPolyInfo_GetActor(&globalCtx->colCtx, floorPolySource);
+        if ((dyna != NULL) && (dyna->actor.id == ACTOR_BG_BDAN_SWITCH)) {
+            if ((((dyna->actor.params) >> 8) & 0x3F) == 0x38) {
+                gSaveContext.inf_table[20] |= 1;
+                return;
+            }
+        }
+    }
+    gSaveContext.inf_table[20] &= ~0x1;
+}
```

Comment:
else instead of early return probably

Reply:
Maybe I'm missing something, but I'm not sure how to easily do that and keep it equivalent. I'd think each if would need an else in that case, and it's not like they can be combined.